bin/main.rb

```diff
@@ -1,73 +1,73 @@
+  puts '        |       |       '
+  puts "    #{board.sq_val[0]}   |   #{board.sq_val[1]}   |   #{board.sq_val[2]}   "
+  puts '        |       |       '
+  puts '        |       |       '
+  puts ' -----------------------'
+  puts '  ⁴     | ⁵     | ⁶     '
+  puts '        |       |       '
+  puts "    #{board.sq_val[3]}   |   #{board.sq_val[4]}   |   #{board.sq_val[5]}   "
+  puts '        |       |       '
+  puts '        |       |       '
+  puts ' -----------------------'
+  puts '  ⁷     | ⁸     | ⁹     '
+  puts '        |       |       '
+  puts "    #{board.sq_val[6]}   |   #{board.sq_val[7]}   |   #{board.sq_val[8]}   "
+  puts '        |       |       '
+  puts '        |       |       '
 end
 
-def draw_board(square_values)
-  puts "     0       1       2   "
-  puts "         |       |       "
-  puts "0    #{square_values[0]}   |   #{square_values[1]}   |   #{square_values[2]}   "
-  puts "         |       |       "
-  puts "  -----------------------"
-  puts "         |       |       "
-  puts "1    #{square_values[3]}   |   #{square_values[4]}   |   #{square_values[5]}   "
-  puts "         |       |       "
-  puts "  -----------------------"
-  puts "         |       |       "
-  puts "2    #{square_values[6]}   |   #{square_values[7]}   |   #{square_values[8]}   "
-  puts "         |       |       "
-end
+print 'Player one name: '
+name = gets.chomp
+player_one = Player.new(name, 'X')
 
-def get_name(num, weapon)
-  puts "Player #{num} name:"
-  name = gets.chomp
-  player_one = Player.new(name, weapon)
-end
+print 'Player two name: '
+name = gets.chomp
+player_two = Player.new(name, 'O')
 
-def switch_player(active_player, player_one, player_two)
-  if active_player == player_one.name
-    active_player = player_two.name
-    active_weapon = player_two.weapon
-  elsif active_player == player_two.name
-    active_player = player_one.name
-    active_weapon = player_one.weapon
-  end
-end
+board = Board.new
 
-player_one = get_name("one", "X")
-player_two = get_name("two", "O")
+puts board.sq_val[0]
 
-square_values = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+game = Game.new(player_one, player_two, board)
 
-active_player = player_one.name
-active_weapon = player_one.weapon
+active_player = player_one
+count = 0
+game_over = false
 
-puts active_player
-puts active_weapon
+while game_over == false
+  draw_board(board)
 
-draw_board(square_values)
-puts "#{active_player}, choose a square: "
-choice = gets.chomp.to_i
-choice -= 1
+  puts "#{active_player.name}, choose a square value: "
+  validation = board.square_play(gets.chomp.to_i - 1, active_player.weapon)
 
-while square_values[choice.to_i] != " "
-  puts "Invalid square. Choose an empty square."
-  choice = gets.chomp
-end
+  while validation[0] == false
+    puts validation[1]
+    validation = board.square_play(gets.chomp.to_i - 1, active_player.weapon)
+  end
 
-square_values[choice.to_i] = active_weapon
+  game_over = game.determine_winner(game_over)
```

Comment:
**[REQUIRED]**
We are not using game_over in the determine_winner anymore. So you can remove the param and parenthesis.